coordinator/src/main/scala/filodb.coordinator/NodeCoordinatorActor.scala

```diff
@@ -22,25 +22,31 @@ import filodb.core.reprojector.Reprojector
  * - Metadata changes (dataset/column changes)
  * - Caching changes to dataset metadata?
  * - Supervising, spinning up, cleaning up DatasetCoordinatorActors
- * - Forwarding new changes (rows) to other NodeCoordinatorActors if they are not local
+ * - Forwarding new changes (rows) to other NodeCoordinatorActors if they are not localØ
  * - Forwarding rows to DatasetCoordinatorActors
  *
  * It is called by local (eg HTTP) as well as remote (eg Spark ETL) processes.
  */
 object NodeCoordinatorActor {
   // Internal messages
   case object Reset
-  case class AddDatasetCoord(dataset: DatasetRef, version: Int, dsCoordRef: ActorRef)
+  case class AddDatasetCoord(originator: ActorRef,
+                             dataset: DatasetRef,
+                             version: Int,
+                             dsCoordRef: ActorRef,
+                             reloadFlag: Boolean)
   case class DatasetCreateNotify(dataset: DatasetRef, version: Int, msg: Any)
+  case object ReloadDCA
 
   def invalidColumns(columns: Seq[String], schema: Column.Schema): Set[String] =
     (columns.toSet -- schema.keys)
 
   def props(metaStore: MetaStore,
             reprojector: Reprojector,
             columnStore: ColumnStore,
-            config: Config): Props =
-    Props(classOf[NodeCoordinatorActor], metaStore, reprojector, columnStore, config)
+            config: Config,
+            actorAddress: Address): Props =
```

Comment:
Sure.  You don't need to create the system object, that exists inside every actor.

In any actor, you can do this:

    context.system

to get at the ActorSystem.  Thus instead of passing in the actorAddress, inside the NCA you can just do this:

    val cluster = Cluster(context.system)
    val selfAddr = cluster.selfAddress